Detectors/TPC/workflow/src/CATrackerSpec.cxx

```diff
@@ -73,9 +86,132 @@ DataProcessorSpec getCATrackerSpec(bool processMC, std::vector<int> const& input
       auto& parser = processAttributes->parser;
       auto& tracker = processAttributes->tracker;
       parser = std::make_unique<ClusterGroupParser>();
-      tracker = std::make_unique<o2::tpc::TPCCATracking>();
-      if (tracker->initialize(options.c_str()) != 0) {
-        throw std::invalid_argument("TPCCATracking initialization failed");
+      tracker = std::make_unique<o2::tpc::GPUCATracking>();
+
+      // Prepare initialization of CATracker - we parse the deprecated option string here,
+      // and create the proper configuration objects for compatibility.
+      // This should go away eventually.
+
+      // Default Settings
+      float solenoidBz = -5.00668; // B-field
+      float refX = 83.;            // transport tracks to this x after tracking, >500 for disabling
+      bool continuous = false;     // time frame data v.s. triggered events
+      int nThreads = 1;            // number of threads if we run on the CPU, 1 = default, 0 = auto-detect
+      bool useGPU = false;         // use a GPU for processing, if false uses GPU
+      bool dump = false;           // create memory dump of processed events for standalone runs
+      char gpuType[1024] = "CUDA"; // Type of GPU device, if useGPU is set to true
+      GPUDisplayBackend* display = nullptr;
+
+      const auto grp = o2::parameters::GRPObject::loadFrom("o2sim_grp.root");
+      if (grp) {
+        LOG(INFO) << "Initializing run paramerers from GRP";
+        solenoidBz *= grp->getL3Current() / 30000.;
+        continuous = grp->isDetContinuousReadOut(o2::detectors::DetID::TPC);
+      } else {
+        LOG(ERROR) << "Failed to initialize run parameters from GRP";
+        // should we call fatal here?
+      }
+
+      // Parse the config string
+      const char* opt = options.c_str();
+      if (opt && *opt) {
+        printf("Received options %s\n", opt);
+        const char* optPtr = opt;
+        while (optPtr && *optPtr) {
```

Comment:
An example of such a parser can be found in the FairMQ channel options parsing
https://github.com/FairRootGroup/FairMQ/blob/3582091b1c76cbe689b8e66813ea71dd5ff55bc5/fairmq/options/FairMQSuboptParser.h#L46
https://github.com/FairRootGroup/FairMQ/blob/ee24144d6158c00a7bc2293e28ca8e4a0c36e9f6/fairmq/options/FairMQSuboptParser.cxx#L32

Reply:
again agreed, just moved it here as first step and in order to be compatible.

Reply:
> key[=value] list, e.g. `refX=83,cont`. That would have the advantage to directly use system tool 

The refX is the reconstruction option (its counterpart in the matching code is at the moment hardcoded), should be eventually extracted from Sandro's ConfigurableParam-derived class (stored in ccdb and can be overridden either from the ini file or from command line, e.g.  ``--configKeyValues MatchingParams.mRefX=...``.
Also, since the field and cont/triggered RO mode must correspond to simulation and digitization (hence will be always extracted from the GRP), there is no much sense in looking for them in the options after having parsed from the GRP.